src/lightning_app/core/app.py

```diff
@@ -184,7 +184,8 @@ def __init__(
     def _update_index_file(self):
         # update index.html,
         # this should happen once for all apps before the ui server starts running.
-        frontend.update_index_file(FRONTEND_DIR, info=self.info, root_path=self.root_path)
+        if self.root_path != "":
```

Comment:
why do we check for this condition?

Reply:
This seems to be a bug. That function isn't meant to be called if root_path isn't provided.